src/client/components/filter-tile/filter-tile.tsx

```diff
@@ -274,54 +277,51 @@ export class FilterTile extends React.Component<FilterTileProps, FilterTileState
     this.setState({ dragPosition: null });
   }
 
+  draggingDimension(): Dimension {
+    const { essence: { dataCube } } = this.props;
+    if (DragManager.isDraggingSplit()) {
+      return dataCube.getDimension(DragManager.draggingSplit().reference);
+    }
+    return DragManager.draggingDimension();
+  }
+
   drop = (e: React.DragEvent<HTMLElement>) => {
     if (!this.canDrop()) return;
     e.preventDefault();
-    const { clicker, essence } = this.props;
-    const { filter, dataCube } = essence;
 
     this.setState({ dragPosition: null });
 
     const dragPosition = this.calculateDragPosition(e);
-    if (!DragManager.isDraggingFilter()) {
-      const dimension = DragManager.draggingDimension();
-      let tryingToReplaceTime = false;
-      if (dragPosition.replace !== null) {
-        const targetClause = filter.clauses.get(dragPosition.replace);
-        tryingToReplaceTime = targetClause && targetClause.reference === dataCube.getTimeDimension().name;
-      }
-      if (dragPosition && !tryingToReplaceTime) {
-        this.addDummy(dimension, dragPosition);
-      }
+
+    if (DragManager.isDraggingFilter()) {
+      this.dropFilter(dragPosition);
```

Comment:
Again, isDraggingFilter condition could be checked internally by DragManger.

Reply:
I don't get that. I have one behaviour for dropping Filter and different for dropping Dimension. I think it should be placed there, but that way I need this condition to decide what to do.